Forta-Bot-Challenge-4/COMP03/src/agent.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  Finding,
+  HandleTransaction,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  getEthersProvider,
+} from "forta-agent";
+import { ethers } from "forta-agent";
+import { SUPPLY_RATE_ABI, SUPPLY_RATE_CONTRACT_ADDRESS } from "./constants";
+
+import { getSupplyRate, checkSupplyRate } from "./utils";
+
+export function provideHandleTransaction(
+  provider: ethers.providers.Provider,
+  supplyRateABI: string,
+  blockNumber: number
+): HandleTransaction {
+  return async function handleTransaction(txEvent: TransactionEvent) {
+    const findings: Finding[] = [];
+
+    if (txEvent.to != SUPPLY_RATE_CONTRACT_ADDRESS.toLowerCase()) {
+      return findings;
+    }
+    const swapTxs = txEvent.filterFunction(supplyRateABI);
```

Comment:
`getSupplyRate` of the `supplyRateABI` is a view function, so this will always be empty. You may use `supply(address asset, uint256 amount)` or monitor events instead. However, I think that a better way of implementing this bot would be to use `handleBlock` and then, once per X blocks, call `getUtilization()` to get the utilization and then call `getSuppleRate(..)` with that

Reply:
Sounds good, I'll go ahead and re-implement this logic with the handleBlock following the logic you outlined above